src/System.IO.Pipelines/Pipe.cs

```diff
@@ -493,29 +478,24 @@ ReadableBufferAwaitable IPipeReader.ReadAsync()
 
         private void Dispose()
         {
-            // TODO: Review throw if not completed?
```

Comment:
Since we're removing the lock here should we throw on other operations if disposed is true? That'll at least make sure that nobody is misusing the thing.